citesphere/src/main/java/edu/asu/diging/citesphere/core/util/model/impl/CitationHelper.java

```diff
@@ -59,37 +60,52 @@ public void updateCitation(ICitation citation, CitationForm form) {
         }
         citation.setAuthors(new HashSet<>());
         if (form.getAuthors() != null) {
-            for (PersonForm personForm : form.getAuthors()) {
-                IPerson person;
-                if (personForm.getId() != null && !personForm.getId().isEmpty()) {
-                    person = authorMap.get(personForm.getId());
-                } else {
-                    person = new Person();
-                }
-                person.setFirstName(personForm.getFirstName());
-                person.setLastName(personForm.getLastName());
-                person.setName(String.join(" ", personForm.getFirstName(), personForm.getLastName()));
-                person.setUri(personForm.getUri());
-                
-                Map<String, IAffiliation> affiliationMap = new HashMap<>();
-                if (person.getAffiliations() != null) {
-                    person.getAffiliations().forEach(a -> affiliationMap.put(a.getId(), a));
-                }
-                person.setAffiliations(new HashSet<>());
-                if (personForm.getAffiliations() != null) {
-                    for (AffiliationForm affiliationForm : personForm.getAffiliations()) {
-                        IAffiliation affiliation;
-                        if (affiliationForm.getId() != null && !affiliationForm.getId().isEmpty()) {
-                            affiliation = affiliationMap.get(affiliationForm.getId());
-                        } else {
-                            affiliation = new Affiliation();
-                        }
-                        affiliation.setName(affiliationForm.getName());
-                        person.getAffiliations().add(affiliation);
+            mapPersonFields(citation, authorMap, form.getAuthors(), "author");
+        }
+
+        Map<String, IPerson> editorMap = new HashMap<>();
+        citation.getEditors().forEach(a -> editorMap.put(a.getId(), a));
+        citation.setEditors(new HashSet<>());
+        if (form.getEditors() != null) {
+            mapPersonFields(citation, editorMap, form.getEditors(), "editor");
+        }
+    }
+
+    @Override
+    public void mapPersonFields(ICitation citation, Map<String, IPerson> personMap,
```

Comment:
is that called from anywhere outside of this class? if not, this should be private.